### DISCUSSION TITLE
Suggestions for retrieving weather data.

### DISCUSSION THREAD

**Discussion by JenJell** [InfiniTimeOrg/InfiniLink#100] [General]
Hello everyone!

Following the release of InfiniTime’s new simple weather service API, I've been working on the code required for its integration. And while I have gotten the app transmitting weather data to the watch, I still don’t know what the best API for retrieving weather data would be. Though, if the number of app users is low enough, we could probably use Apple's WeatherKit. Anyway, if anyone has any suggestions they would like to share, that would be appreciated.

**Comment by ocdtrekkie** [comment]
If it can pull from Apple's Weather data on device, and send to my watch, that's preferable: No new location data about me goes to the Internet.

If an alternative service is used and connects from my phone, the risk of tracking becomes a concern. I'd much rather set the location to monitor in the app (say, my town) rather than have the app piping out my current actual location to a web service.

**Comment by liamcharger** [comment]
@JenJell Thanks for all your work! While we can use WeatherKit, I think something like [this](https://www.weather.gov/documentation/services-web-api) _free_ API may be better.

We can get the device's location and then fetch the data accordingly. For [example](https://api.weather.gov/points/39.7456,-97.0892).

**Reply by JenJell** [reply]
Using the National Weather Service API sounds like a good idea to me. Especially given that there doesn’t seem to be a monthly rate limit for it. However, I’ll probably still need to look into a secondary option to fall back on for anyone outside of the United States.

**Reply by liamcharger** [reply]
Ah, that's true, I didn't think about people outside of the US not being able to use it. WeatherKit may be our best option, then, except for it only being available in iOS 16 and above.

**Reply by liamcharger** [reply]
[WeatherApi](https://www.weatherapi.com/) could be an option. There's a million free calls a month, so that should be more than enough. The API responses look good, and the data is accurate.

**Reply by JenJell** [reply]
That looks like it would work well. We can implement that as a fall back option for when the app can't access the NWS API. That way if a user is located outside of the United States, the app still has something to fall back on.

**Reply by liamcharger** [reply]
Yes, sounds good! Thanks again for your contributions!

**Comment by liamcharger** [comment]
@JenJell I've created a weather branch with your implementation, that way we could experiment with different APIs if we need to.

**Comment by liamcharger** [comment]
I'm going to close this discussion now because the weather service has been implemented.
